### PR TITLE
Fix #615 - Use default config instead of env as connection

### LINKED ISSUE #615: Use default config instead of env as connection
This issue relates to https://github.com/stancl/tenancy/issues/164 

The idea is to drop the double `env()` call in package config and use package env or `null` as default value. In model construct we use the `database.default` as fallback value.

PS: I would prefer null coalescing operator over function nesting.
```php
// good
config('key') ?? config('key') ?? 'default';
config('key') ?? config('key', 'default');

// bad
config('key', config('key', 'default'));
```

### PR DESCRIPTION
**Pull request by micc83**
fixes #615

### CODE REVIEW

File: tests/CustomDatabaseConnectionActivityModelTest.php [hunk 1]
```diff
@@ -26,6 +26,16 @@ public function it_uses_a_custom_database_connection()
...
     }
 
+    /** @test */
+    public function it_uses_the_default_database_connection_when_the_one_from_configuration_is_null()
+    {
+        config()->set('activitylog.database_connection', null);
+
+        $model = new Activity();
+
+        $this->assertInstanceOf('Illuminate\Database\SQLiteConnection', $model->getConnection());
```
Comment (stancl) [inline tests/CustomDatabaseConnectionActivityModelTest.php]: I suggest comparing the connection name or the database name, rather than just the type.

**Comment by stancl** [comment]
Looks good to me.

File: tests/CustomDatabaseConnectionActivityModelTest.php (hunk 1 above)
Comment (micc83) [reply tests/CustomDatabaseConnectionActivityModelTest.php]: @stancl`$model->getConnectionName()` is `null` when not set externally. I think that's the desired behaviour, unless I'm missing something, and that's why I'm looking into the resulting connection.
```php
    // Illuminate\Database\ConnectionResolver
    public function connection($name = null)
    {
        if (is_null($name)) {
            $name = $this->getDefaultConnection();
        }

        return $this->connections[$name];
    }
```

File: .gitignore [hunk 2]
```diff
@@ -2,3 +2,4 @@ build
 composer.lock
 vendor
 .phpunit.result.cache
+.idea
```
[the remark this answers is not in the document]
Comment (micc83) [reply .gitignore]: Sure, in our company we usually includes them to avoid new hired forgetting about em. Fixed it.

**Comment by Gummibeer** [comment]
https://github.com/spatie/laravel-activitylog/releases/tag/3.9.1

**Pull request merged**